data/env/snapd.sh.in

```diff
@@ -1,14 +1,17 @@
 #!/bin/sh --this-shebang-is-just-here-to-inform-shellcheck--
 
 # Expand $PATH to include the directory where snappy applications go.
-if [ "${PATH#*/snap/bin}" = "${PATH}" ]; then
+if [ -n "${PATH##*@SNAP_MOUNT_DIR@/bin*}" ]; then
```

Comment:
Nice!

Reply:
I wasn't sure so I checked that works in dash as well as bash.

Reply:
That looks like it's picking up anything that is prefixed by "bin", instead of just the exact "/snap/bin" string. Is it not?

Reply:
That looks like it's picking up anything that is prefixed by "bin", instead of just the exact "/snap/bin" string. Is it not?

Reply:
Maybe `echo $PATH | /usr/bin/grep -qE '@SNAP_MOUNT_DIR@/bin(:|$)'` would be more appropriate then.

Reply:
It's actually taking anything that contains "/snap/bin", including subdirectories if already set. The other way, would be to do what @bboozzoo suggested for both cases, but it means subprocessing at startup time.

Reply:
Somewhat brute force:
```
if [ -n "${PATH##*@SNAP_MOUNT_DIR@/bin}" -o -n "${PATH##*@SNAP_MOUNT_DIR@/bin:*}" ]; ..
```
no subprocess though

Reply:
Correct, it's a little bit redundant and unclear to the reader I guess, but maybe that's better?

Reply:
Uhh `s/-o/-a/`